[Site Isolation] Fix more cases where we unnecessarily skip code on out-of-process frames

charliewolfe opened the pull request:
#### 008862a638f51cd93380739533aaedaee0a3916f
<pre>
[Site Isolation] Fix more cases where we unnecessarily skip code on out-of-process frames
<a href="https://bugs.webkit.org/show_bug.cgi?id=266007">https://bugs.webkit.org/show_bug.cgi?id=266007</a>
<a href="https://rdar.apple.com/119326844">rdar://119326844</a>

Reviewed by Pascoe.

* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::arePluginsEnabled):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::arePluginsEnabled): Deleted.
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/PageOverlay.cpp:
(WebCore::PageOverlay::viewToOverlayOffset const):
(WebCore::PageOverlay::mouseEvent):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::resolveChildFrameWithNodeHandle):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageCopyContextMenuAtPointInWindow):
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
(WebKit::WebFoundTextRangeController::rectsForTextMatchesInRect):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::setAlwaysShowsHorizontalScroller):
(WebKit::WebPage::setAlwaysShowsVerticalScroller):
(WebKit::WebPage::canShowMIMEType const):

Canonical link: <a href="https://commits.webkit.org/271723@main">https://commits.webkit.org/271723@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3a9197b2cbe02b6b666ec8f3a512ea7957938e4f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29415 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8080 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30739 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31965 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26685 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30015 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10216 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5372 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26682 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29688 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6712 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25142 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5778 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5905 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26217 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33306 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26834 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26632 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32126 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5868 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4049 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29905 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7578 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6996 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6395 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6392 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pascoej commented on Source/WebCore/page/FrameTree.cpp:
```diff
@@ -396,11 +396,11 @@ Frame* FrameTree::firstRenderedChild() const
         return nullptr;
 
-    if (auto* localChild = dynamicDowncast<LocalFrame>(child); localChild && localChild->ownerRenderer())
+    if (auto* localChild = child; localChild && localChild->ownerRenderer())
```
```suggestion
    if (child->ownerRenderer())
        return child;
```

pascoej commented on Source/WebCore/page/FrameTree.cpp:
```diff
@@ -396,11 +396,11 @@ Frame* FrameTree::firstRenderedChild() const
 
     while ((child = child->tree().nextSibling())) {
-        if (auto* localChild = dynamicDowncast<LocalFrame>(child); localChild && localChild->ownerRenderer())
+        if (auto* localChild = child; localChild && localChild->ownerRenderer())
```
ditto

pascoej commented on Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
```diff
@@ -527,7 +527,7 @@ void WebAutomationSessionProxy::resolveChildFrameWithNodeHandle(WebCore::PageIde
     }
 
-    auto* coreFrameFromElement = dynamicDowncast<WebCore::LocalFrame>(downcast<WebCore::HTMLFrameElementBase>(*coreElement).contentFrame());
+    auto* coreFrameFromElement = downcast<WebCore::HTMLFrameElementBase>(*coreElement).contentFrame();
```
We could downcast to HTMLFrameOwnerElement here instead (one level up but still has contentFrame.)

I think this is just a nit though.

charliewolfe committed:
[Site Isolation] Fix more cases where we unnecessarily skip code on out-of-process frames
https://bugs.webkit.org/show_bug.cgi?id=266007
rdar://119326844

Reviewed by Pascoe.

* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::arePluginsEnabled):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::arePluginsEnabled): Deleted.
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/PageOverlay.cpp:
(WebCore::PageOverlay::viewToOverlayOffset const):
(WebCore::PageOverlay::mouseEvent):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::resolveChildFrameWithNodeHandle):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageCopyContextMenuAtPointInWindow):
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
(WebKit::WebFoundTextRangeController::rectsForTextMatchesInRect):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::setAlwaysShowsHorizontalScroller):
(WebKit::WebPage::setAlwaysShowsVerticalScroller):
(WebKit::WebPage::canShowMIMEType const):

Canonical link: https://commits.webkit.org/271723@main
